src/lib.rs

```diff
@@ -405,17 +424,93 @@ impl purple::CommandHandler for PurpleICQ {
         command: &str,
         args: &[&str],
     ) -> PurpleCmdRet {
-        log::error!(
-            "cmd_func: conv={} cmd={} args={:?}",
+        log::debug!(
+            "command: conv={} cmd={} args={:?}",
             conversation.get_title().unwrap_or("unknown"),
             command,
             args
         );
-        PurpleCmdRet::PURPLE_CMD_RET_OK
+        match command {
+            commands::HISTORY => self.command_history(conversation, args),
+            _ => {
+                log::error!("Unknown command: {}", command);
+                PurpleCmdRet::PURPLE_CMD_RET_FAILED
+            }
+        }
     }
 }
 
 impl PurpleICQ {
+    fn command_history(&mut self, conversation: &mut Conversation, args: &[&str]) -> PurpleCmdRet {
+        log::debug!("command_history");
+
+        if args.len() != 1 {
+            log::error!(
+                "command_history: Unsupported number of args. Got {}",
+                args.len()
+            );
+            return PurpleCmdRet::PURPLE_CMD_RET_FAILED;
+        }
+        let count = {
+            let input = match args[0].parse::<i32>() {
```

Comment:
nit, could make it an `u32` to avoid negative counts

Reply:
Sure, done.